src/components/Pokemon.jsx

```diff
@@ -27,56 +40,132 @@ const Pokemon = ({ url }) => {
+      return false;
+    }
+    return true;
+  };
+
+  const displayAbility = () => {
+    if (
+      abilityFilters !== 'all' &&
+      !infos.abilities
+        .map((element) => element.ability.name)
+        .includes(abilityFilters)
+    ) {
+      return false;
+    }
+    return true;
+  };
+
+  const displayHeight = () => {
+    if (heightFilters === 'small') {
+      return infos.height < 10;
+    }
+    if (heightFilters === 'medium') {
+      return infos.height >= 10 && infos.height <= 20;
+    }
+    if (heightFilters === 'big') {
+      return infos.height > 20;
+    }
+    return true;
+  };
+
+  const displayWeight = () => {
+    if (weightFilters === 'light') {
+      return infos.weight < 1000;
+    }
+    if (weightFilters === 'medium') {
+      return infos.weight >= 1000 && infos.weight <= 3000;
+    }
+    if (weightFilters === 'heavy') {
+      return infos.weight > 3000;
+    }
+    return true;
+  };
+
+  useEffect(() => {
+    setDisplay(
+      displayType() && displayAbility() && displayHeight() && displayWeight()
+    );
+  }, [typesFilters, abilityFilters, heightFilters, weightFilters]);
+
   return (
-    <li
-      className="pokemon"
-      style={{ backgroundColor: colorTypes[infos.types[0].type.name] }}
-    >
-      <Link to={`/${infos.name}`}>
-        <img
-          className="pokemon-image"
-          src={infos.sprites.other['official-artwork'].front_default}
-          alt="pokemon avatar"
-        />
-        <div className="pokemon-infos">
-          <h2 className="pokemon-name">
-            {infos.name.charAt(0).toUpperCase() + infos.name.slice(1)}
-          </h2>
-          <p className="pokemon-id">
-            #
-            {infos.id.toLocaleString('en-US', {
-              minimumIntegerDigits: 3,
-              useGrouping: false,
-            })}
-          </p>
-          <p className="pokemon-types">
-            {infos.types.map((element) => element.type.name).join(' - ')}
-          </p>
-        </div>
-        <div className="arrow-container">
-          <img className="arrow" src="/img/arrow.svg" alt="right arrow" />
-        </div>
-      </Link>
-    </li>
+    <>
+      {loading && <div className="loading" />}
+      {error && <div className="error" />}
+      {!loading && !error && display && (
+        <li className={`pokemon ${infos.types[0].type.name}`}>
+          <Link to={`/${infos.name}`}>
+            <img
+              className="pokemon-image"
+              src={infos.sprites.other['official-artwork'].front_default}
+              alt="pokemon avatar"
+            />
+            <div className="pokemon-infos">
+              <h2 className="pokemon-name">
+                {infos.name.charAt(0).toUpperCase() + infos.name.slice(1)}
+              </h2>
+              <p className="pokemon-id">
+                #
+                {infos.id.toLocaleString('en-US', {
+                  minimumIntegerDigits: 3,
+                  useGrouping: false,
+                })}
```

Comment:
Did you really need this ? o.O

Reply:
Yes for a better display of id on the Pokemon component, I have move this in the .then of axios call